### ISSUE TITLE
Matcher on website doesn't show matches correctly on e.g. IPv6

### ISSUE THREAD

**Issue by moviuro** [geongeorge/i-hate-regex#6]
Hi,

The following IPs do match the given regexp if used in `grep(1)` but not entirely on the [website](http://ihateregex.io/expr/ipv6#). I use bold (double star) in this issue to show which part were highlighted. https://github.com/geongeorge/i-hate-regex/blob/dc4299ebbe6b4c6ff0cf826d1f051e50f3133269/static/regexdata.json#L176

Those should completely match:

- **2001:db8::**1
- **123::**456:789
- **FE80:0000:0000:0000:0202:B3FF:FE1E:8329**%eth0
- **fe80::**aaaa:bbbb:eeee:1111%eth0

**Comment by geongeorge** [comment]
Is this related to javascript regex engine and the one in grep?

**Comment by moviuro** [comment]
Yes, probably. With the same regex and the same test strings, matches differ when they shouldn't (only partial matches in JavaScript, and complete matches in `grep(1)`). I'm however unable to help any further: though I know regex, I know nothing of JS...

**Comment by geongeorge** [comment]
Somebody submitted this via google forms and claimed it's better:
```
^(?:[A-Fa-f0-9]{1,4}:){7}[A-Fa-f0-9]{1,4}$
```

Can anybody confirm this? @moviuro

**Comment by geongeorge** [comment]
> Yes, probably. With the same regex and the same test strings, matches differ when they shouldn't (only partial matches in JavaScript, and complete matches in `grep(1)`). I'm however unable to help any further: though I know regex, I know nothing of JS...

Can you also check this?
https://regex101.com/r/tEXuwr/1

**Comment by moviuro** [comment]
> Somebody submitted this via google forms and claimed it's better:
> 
> ```
> ^(?:[A-Fa-f0-9]{1,4}:){7}[A-Fa-f0-9]{1,4}$
> ```
> 
> Can anybody confirm this? @moviuro

This can't match IPv4 compatible addresses (e.g. `::FFFF:222.1.41.90`).

The regex101 link has the same issue as the i-hate-regex: it only does partial matches when it should find full matches.

**Comment by geongeorge** [comment]
If you can provide a source for specification for these address types or write it down. 
I'll build the expression for javascript

**Comment by jjimenezshaw** [comment]
https://en.wikipedia.org/wiki/IPv6#IPv4-mapped_IPv6_addresses
